### PR TITLE
hashid_table_name

### PR DESCRIPTION
**Pull request by gsdean**
Allows for table to be renamed and not break existing ids

### CODE REVIEW

File: lib/hashid/rails.rb [hunk 1]
```diff
@@ -84,10 +84,13 @@ def find_by_hashid!(hashid)
         find_by!(id: decode_id(hashid, fallback: false))
       end
 
+      def hashid_table_name
```
Comment (jcypret) [inline lib/hashid/rails.rb]: Is there a compelling reason for making this public? I'd prefer it kept private since it's for internal use by the gem.

Comment (gsdean) [reply lib/hashid/rails.rb]: Not necessarily `public`, although, I think `protected` might be a better choice as the idea is that you override the method when/if you need to change the original class name.

**Comment by bjeanes** [comment]
The fact that the model name is incorporated into the Hashids config is the main reason I rolled my own instead of using this. I'd love to see this so that I could remove some custom code, though I would encourage not tying this configuration concept to "table" and instead just generally allow a different "seed"/"salt" per-model to be mixed in.

My own version of this gem allows for something like:

``` ruby
class Author < ApplicationRecord
  # `salt`, `alphabet`, etc fallback to global but `pepper` is required
  hashes_ids pepper: 'authors'
end
```

**Comment by jcypret** [comment]
I like the idea of changing the model salt to be independent of table. An original reason for the inclusion was ensuring that same integer ids where not identical across models. Maybe making the table name the default, but allowing model level config would be a good compromise.

I'm going to close this PR in favor of a model level config.

**Comment by gsdean** [comment]
Sorry, I’m confused. Is that not exactly what this did?

**Comment by jcypret** [comment]
> Sorry, I’m confused. Is that not exactly what this did?

Yes, this PR addresses specifically overriding table name, but after reviewing the other PRs and issues, it has become more apparent that we need to move to a model level config. The table name isn't the only config that needs to be overridden at the model level and I'd rather not set a precedence for adding new override methods for each config.

So instead of adding a `hashid_table_name` method to any models that need customization, I'd prefer we made each model configurable in a single config line rather than across multiple private methods.  Building on what @bjeanes outlined, maybe something like the below example where `pepper` is set to the `table_name` by default, but allows overring in the model along with other options:

```ruby
class RenamedModel < ApplicationRecord
  hashids pepper: "old_table_name", min_hash_length: 8
end
```

Tracking a separate issue here: https://github.com/jcypret/hashid-rails/issues/60

**Pull request closed without merging**